planner/core/rule_eliminate_projection.go

```diff
@@ -137,6 +138,16 @@ func (pe *projectionEliminater) eliminate(p LogicalPlan, replace map[string]*exp
 		}
 	}
 	p.replaceExprColumns(replace)
+	if len(p.Children()) > 0 {
```

Comment:
I think it would be more clear to write this as:
```go
if isProj {
    if child, ok := p.Children()[0].(*LogicalProjection);  ok {
    }
}
```

Reply:
Fixed